lib/lightning_web/live/credential_live/index.html.heex

```diff
@@ -52,15 +67,28 @@
           <.td>
             <%= if !credential.scheduled_deletion do %>
               <span>
-                <.link navigate={
-                  Routes.credential_edit_path(@socket, :edit, credential)
+                <.link phx-click={
+                  show_modal("edit-credential-#{credential.id}-modal")
                 }>
                   Edit
                 </.link>
               </span>
               |
             <% end %>
             <.delete_action socket={@socket} credential={credential} />
+            <.live_component
+              id={"edit-credential-#{credential.id}-modal"}
+              module={LightningWeb.CredentialLive.FormComponent}
+              action={:edit}
+              credential_type={@selected_credential_type}
+              credential={credential}
+              current_user={@current_user}
+              projects={@projects}
+              project={nil}
+              show_project_credentials={true}
+              can_create_project_credential={true}
+              return_to={~p"/credentials"}
+            />
```

Comment:
I'm not that keen on rendering components for every item, it seems a bit wasteful. I don't think we should do anything about it now given how infrequently its accessed but we should think about lowering the cost of rendering/diffing at every point; especially in lists.

Reply:
@stuartc I do agree. I also am not a big fan of this. Seems like the most reasonable thing to do now, is keep as it is and think about a better approach earlier January.